src/api/index.js

```diff
@@ -24,15 +24,24 @@ export function getAPIRoot() {
 
 const apiRoot = getAPIRoot();
 
-export function getAPI(type, { name = '', namespace } = {}) {
+export function getAPI(type, { name = '', namespace } = {}, queryParams) {
   return [
     apiRoot,
     '/v1/namespaces/',
     encodeURIComponent(namespace),
     '/',
     type,
     '/',
-    encodeURIComponent(name)
+    encodeURIComponent(name),
+    queryParams
```

Comment:
browsers have native support for building the query string, including encoding as needed:

```js
queryParams ? `?${new URLSearchParams(queryParams).toString()}` : ''
```

e.g.
![image](https://user-images.githubusercontent.com/2829095/57472500-79ba9a80-7285-11e9-858d-7f8395db076a.png)